src/api/spec/models/workflow/step/link_package_step_spec.rb

```diff
@@ -0,0 +1,310 @@
+        source_project: package.project.name,
+        source_package: package.name
+      }
+    end
+
+    it { expect { subject.call }.to(change(Project, :count).by(1)) }
+    it { expect { subject.call }.to(change(Package, :count).by(2)) }
+    it { expect(subject.call.project.name).to eq(target_project_name) }
+    it { expect { subject.call }.to(change(EventSubscription.where(eventtype: 'Event::BuildFail'), :count).by(1)) }
+    it { expect { subject.call }.to(change(EventSubscription.where(eventtype: 'Event::BuildSuccess'), :count).by(1)) }
+    it { expect { subject.call.source_file('_branch_request') }.not_to raise_error }
+    it { expect(subject.call.source_file('_branch_request')).to include('123') }
+    it { expect { subject.call.source_file('_link') }.not_to raise_error }
+    it { expect(subject.call.source_file('_link')).to eq('<link project="foo_project" package="bar_package"/>') }
+    it { expect(subject.call.project.packages.map(&:name)).to include('_project') }
+    it { expect { subject.call.project.packages.find_by(name: '_project').source_file('_service') }.not_to raise_error }
+  end
+
+  RSpec.shared_context 'successful update event when the linked_package already exists' do
+    let(:step_instructions) do
+      {
+        source_project: package.project.name,
+        source_package: package.name
+      }
+    end
+
+    # Emulate the linked project/package and the subcription created in a previous new PR/MR event
+    let!(:linked_project) { create(:project, name: target_project_name, maintainer: user) }
+    let!(:linked_package) { create(:package_with_file, name: package.name, project: linked_project) }
+
+    ['Event::BuildFail', 'Event::BuildSuccess'].each do |build_event|
+      let!("event_subscription_#{build_event.parameterize}") do
+        EventSubscription.create(eventtype: build_event,
+                                 receiver_role: 'reader',
+                                 user: user,
+                                 channel: :scm,
+                                 enabled: true,
+                                 token: token,
+                                 package: linked_package,
+                                 payload: creation_payload)
+      end
+    end
+
+    before do
+      package.save_file({ file: existing_branch_request_file, filename: '_branch_request' })
+    end
+
+    it { expect { subject.call }.to(change(Package, :count).by(0)) }
+    it { expect { subject.call.source_file('_branch_request') }.not_to raise_error }
+    it { expect { subject.call.source_file('_link') }.not_to raise_error }
+    it { expect(subject.call.source_file('_link')).to eq('<link project="foo_project" package="bar_package"/>') }
+
+    it 'updates _branch_request file including new commit sha' do
+      expect(subject.call.source_file('_branch_request')).to include('456')
+    end
+
+    it { expect { subject.call }.to(change(EventSubscription.where(eventtype: 'Event::BuildFail'), :count).by(0)) }
```

Comment:
This is easier to understand. This comment applies to all similar examples.

```suggestion
    it { expect { subject.call }.not_to(change(EventSubscription.where(eventtype: 'Event::BuildFail'), :count)) }
```

Reply:
Done